### PR TITLE
Kong v0.12.2 support and flatten ports values

### PR DESCRIPTION
**Pull request by shashiranjan84**
- appVersion updated to 0.12.2
- ports input structure flatten

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

**Comment by shashiranjan84** [comment]
Problem is with Kong docker image, we working on it.

**Comment by shashiranjan84** [comment]
@unguiculus test failing because NodePort not available, even after removing default value for `NodePort`. Please advise.

File: stable/kong/values.yaml [hunk 1]
```diff
@@ -61,7 +57,7 @@ resources: {}
 readinessProbe:
   httpGet:
     path: "/status"
-    port: admin-ssl
+    port: admin
     scheme: HTTPS
```
Comment (cilindrox) [inline stable/kong/values.yaml:61]: @shashiranjan84 shouldn't the scheme depend on `values.useTLS` too?

File: stable/kong/values.yaml [hunk 2]
```diff
@@ -73,7 +69,7 @@ readinessProbe:
 livenessProbe:
   httpGet:
     path: "/status"
-    port: admin-ssl
+    port: admin
     scheme: HTTPS
```
Comment (cilindrox) [inline stable/kong/values.yaml:73]: same as above for `scheme`

File: stable/kong/values.yaml (hunk 1 above)
Comment (shashiranjan84) [reply stable/kong/values.yaml:61]: @cilindrox its the default value and user would change the scheme if they don't `useTLS`

File: stable/kong/templates/deployment.yaml [hunk 3]
```diff
@@ -28,14 +28,28 @@ spec:
...
         - name: KONG_PROXY_LISTEN_SSL
-          value: 0.0.0.0:{{ default 8443 .Values.proxy.https.containerPort }}
+          value: 0.0.0.0:{{ .Values.proxy.containerPort }}
+        - name: KONG_SSL
+          value: "on"
+        {{- else }}
+        - name: KONG_PROXY_LISTEN
+          value: 0.0.0.0:{{ .Values.proxy.containerPort }}
+        - name: KONG_SSL
+          value: "off"
```
Comment (marcostvz) [inline stable/kong/templates/deployment.yaml:51]: What's the purpose of those env vars `KONG_ADMIN_SSL` and `KONG_SSL`?

For the rest, I think it's a good PR, if this is accepted it I'll close #4082 :)

Comment (shashiranjan84) [reply stable/kong/templates/deployment.yaml:51]: [KONG_SSL](https://getkong.org/docs/0.12.x/configuration/#ssl) and [KONG_ADMIN_SSL](https://getkong.org/docs/0.12.x/configuration/#admin_ssl) are Kong configurations.

Comment (shashiranjan84) [reply stable/kong/templates/deployment.yaml:51]: I am waiting for @unguiculus to approve the PR.

**Comment by shashiranjan84** [comment]
thanks @unguiculus.

**Pull request merged**